test/unit/descriptors.spec.ts

```diff
@@ -0,0 +1,92 @@
+import * as util from "../util";
+
+const trueFalseUndefTests = [[true], [false], [undefined]] as const;
+
+describe("Object.defineProperty", () => {
+    test.each(trueFalseUndefTests)("writable (%p)", value => {
+        util.testFunction`
+            const foo = { bar: true };
+            Object.defineProperty(foo, "bar", { writable: ${value} });
+            try { foo.bar = false } catch {}
```

Comment:
Is there any test that makes sure that it throws?

Reply:
Modified the test to capture errors in the result